crates/framework/src/recording_index.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    fs::File,
+    io::{Read, Seek, SeekFrom},
+    path::Path,
+    time::SystemTime,
+};
+
+use bincode::deserialize_from;
+use color_eyre::{eyre::WrapErr, Result};
+
+#[derive(Debug)]
+pub struct RecordingIndex {
+    file: File,
+    frames: Vec<RecordingFrameMetadata>,
+}
+
+impl RecordingIndex {
+    pub fn read_from(recording_file: impl AsRef<Path>) -> Result<Self> {
+        let file = File::open(&recording_file)
+            .wrap_err_with(|| format!("failed to open {}", recording_file.as_ref().display()))?;
+        Self::collect_frames(file).wrap_err("failed to collect frames")
+    }
+
+    fn collect_frames(mut recording_file: File) -> Result<Self> {
```

Comment:
Yes, true, this is important. Are you fine with moving this into next iteration?

Reply:
Do you think we can finish the next iteration before German Open?
I think this would be important to have by then. We can always retest in the lab, but having unusable replays after a game would be pretty disappointing.

Reply:
Sure, we can work on that before German Open. That should not be complicated to make robust. But it is a next maturity level and it should work well enough without it for the moment. In addition, branching the next iteration from `main` also means that it must not  be me to implement it because I'm losing interest in this topic after so long.